packages/theme-check-common/src/checks/liquid-free-settings/index.ts

```diff
@@ -0,0 +1,68 @@
+import { JSONNode, LiquidCheckDefinition, Severity, SourceCodeType } from '../../types';
+import { toJSONAST } from '../../to-source-code';
+import { visit } from '../../visitor';
+import { LiteralNode } from 'json-to-ast';
+
+export const LiquidFreeSettings: LiquidCheckDefinition = {
+  meta: {
+    code: 'LiquidFreeSettings',
+    name: 'Check for liquid free settings values',
+    docs: {
+      description: 'Ensures settings values are liquid free.',
+      recommended: true,
+      url: 'https://shopify.dev/docs/themes/tools/theme-check/checks/liquid-free-settings',
+    },
+    type: SourceCodeType.LiquidHtml,
+    severity: Severity.WARNING,
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    return {
+      async LiquidRawTag(node) {
+        if (node.name !== 'schema' || node.body.kind !== 'json') {
+          return;
+        }
+
+        const jsonString = node.source.slice(
+          node.blockStartPosition.end,
+          node.blockEndPosition.start,
+        );
+
+        const jsonFile = toJSONAST(jsonString);
+        if (jsonFile instanceof Error) return;
+
+        visit<SourceCodeType.JSON, void>(jsonFile, {
+          Property(schemaNode, ancestors) {
+            if (isInArrayWithParentKey(ancestors, 'settings') && isLiteralNode(schemaNode.value)) {
+              const value = schemaNode.value.value;
```

Comment:
Nit: You could destructure both `value` and `loc` out of `schemaNode.value` so your message indicies could read a bit more concise down the line.
```suggestion
              const {value, loc} = schemaNode.value.value;
```